lib/flags/initialization.ts

```diff
@@ -0,0 +1,87 @@
+import { logMessage } from "@lib/logger";
+import { getRedisInstance } from "../integration/redisConnector";
+import initialFlags from "./default_flag_settings.json";
+
+const cacheAvailable: boolean = process.env.APP_ENV !== "test" && Boolean(process.env.REDIS_URL);
+
+const checkAll = async () => {
+  const redis = await getRedisInstance();
+  const keys = await redis.smembers("flags");
+  return checkMulti(keys);
+};
+
+const checkMulti = async (keys: string[]) => {
+  const redis = await getRedisInstance();
+  if (keys.length === 0) return {};
+
+  const values = await redis.mget(keys.map((key) => `flag:${key}`));
+
+  const mapped = keys.reduce((acc, key, index) => {
+    acc.set(key, values[index] === "1");
+    return acc;
+  }, new Map());
+
+  return Object.fromEntries(mapped);
+};
+
+const removeFlag = async (key: string) => {
+  const redis = await getRedisInstance();
+  await redis.srem("flags", key);
+  await redis.del(`flag:${key}`);
+};
+
+const createFlag = async (key: string, value: boolean) => {
+  const redis = await getRedisInstance();
+  await redis
+    .multi()
+    .sadd("flags", key)
+    .set(`flag:${key}`, value ? "1" : "0")
+    .exec();
+};
+export const initiateFlags = async () => {
+  logMessage.info("Running flag initiation");
+  if (!cacheAvailable) {
+    process.env.APP_ENV !== "test" &&
+      logMessage.info("Application is TEST mode, skipping flag initiation");
+    Boolean(process.env.REDIS_URL) ??
+      logMessage.info("No REDIS_URL environment variable found, skipping flag initiation");
+  }
```

Comment:
We probably don't need this check here since it is already done in the `instrumentation.ts` code.

However, if we do need it, we are missing a `return;` at the end to actually skip the operation.

Reply:
Thank you for the catch about the missing `return`.  My brain is turning to mush.